mlflow/transformers.py

```diff
@@ -1461,12 +1444,24 @@ def _predict(self, data):
             conversation_output = self.pipeline(self._conversation)
             return conversation_output.generated_responses[-1]
         elif isinstance(data, dict):
-            raw_output = self.pipeline(**data)
+            if self.inference_config:
+                raw_output = self.pipeline(**data, **self.inference_config)
+            else:
+                raw_output = self.pipeline(**data)
         else:
-            raw_output = self.pipeline(data)
+            if self.inference_config:
+                raw_output = self.pipeline(data, **self.inference_config)
+            else:
+                raw_output = self.pipeline(data)
 
         # Handle the pipeline outputs
-        if isinstance(self.pipeline, transformers.FillMaskPipeline):
+        if type(self.pipeline).__name__ in self._supported_custom_generator_types or isinstance(
+            self.pipeline, transformers.TextGenerationPipeline
```

Comment:
When loading, since the custom new pipeline type is not installed, the super class will be loaded for Dolly. Dolly's InstructionalTextGenerationPipeline inherits from TextGenerationPipeline, so it still works just fine when loaded through the pyfunc loader. We just need to make sure that we're parsing this output when loaded as the super class.